utilities/cwp/custom-war-packager.inc

```diff
@@ -0,0 +1,90 @@
+# function to download the CWP jar
+# $1: Working directory
+# $2: CWP version to use
+download_cwp() {
+
+	if [ "$#" -gt 1 ]
+	then
+		cwp_jar_file="$1/.cwp-build/cwp-cli-$2.jar"
+		if [ "$#" -eq 4 ]
+		then
+            image_tag="$4"
+        else
+            image_tag="jenkins-experimental/jenkinsfile-runner-test"
+        fi
+
+		rm -rf "$1/.cwp-build"
+		mkdir -p "$1/.cwp-build"
+		wget -O "$cwp_jar_file" https://repo.jenkins-ci.org/releases/io/jenkins/tools/custom-war-packager/custom-war-packager-cli/$2/custom-war-packager-cli-$2-jar-with-dependencies.jar
+
+		echo "CWP downloaded as $cwp_jar_file"
+
+		return 0
+	else
+		echo "Error. Missing parameters:"
+		echo "   Working directory"
+		echo "   CWP version to use"
+		return 1
+	fi
+
+}
+
+# function to generate the docker image using CWP using a downloaded jar
+# $1: Working directory
+# $2: Path to cwp.jar
+# $3: Jenkins version
+# $4: Path to packager config file
+# $5: Tag for the docker image. "jenkins-experimental/jenkinsfile-runner-test" by default
+generate_docker_image_from_jar() {
+
+	if [ "$#" -gt 3 ]
+   	then
+		if [ "$#" -eq 5 ]
+   	    then
+   	        image_tag="$5"
+   	    else
+   	        image_tag="jenkins-experimental/jenkinsfile-runner-test"
+   	    fi
+
+   	    java -jar $2 -configPath $4 -tmpDir "$1/out/tmp/" -version $3
+        docker build -t $image_tag -f "$1/out/tmp/output/Dockerfile" "$1/out/tmp/output/"
```

Comment:
It can probably be done later, but the image could be already built through CWP so this command is not necessary.

Reply:
What do you mean?

Reply:
That if CWP configuration says so, CWP will automatically build the docker image.

Reply:
As discussed offline, I'll split this function into two:
- `execute_cwp_jar` which execute the jar. The docker image will only be generated if and only if the  yaml configuration is configuring its building.
- `execute_cwp_jar_and_build_image` which execute the jar file and the `docker build` command.